### DISCUSSION TITLE
Questions regarding the "SSL settings" fields

### DISCUSSION THREAD

**Discussion by nguyenthaichinghia** [tilln/jmeter-iso8583#96] [Q&A]
> SSL Settings: For SSL/TLS connections, the Keystore File (protected with Keystore Password for the file and Key Password for a private key) should contain:
> Client mode: the server's public certificate to trust and, optionally, the client certificate to send for authentication (mutual SSL).

I'm a student trying to learn about ISO 8583. I'm trying to use mTLS to send the message, in the Keystore File i should have the receiving server's certificate, my certificate and my private key right? But how do the plugin know which certificate is mine and which is the server's? Do i have to define the alias somewhere? Also why do i need to have the server's certificate in the Keystore File, is the certificate there to be compared to the certificate sent by the server when the connection is invoked? Any help is greatly appreciated, and sorry for my bad English!

**Comment by tilln** [comment]
> in the Keystore File i should have the receiving server's certificate, my certificate and my private key right?

Yes, correct: one entry for the server cert, without private key, and one for the client, with private key.

> But how do the plugin know which certificate is mine and which is the server's? Do i have to define the alias somewhere?

No need to define an alias.
If only one of the two certs in the store has a private key there should not be any ambiguity for choosing the client cert.

This plugin only passes the keystore details to jPOS which are then used presumably [here](https://github.com/jpos/jPOS/blob/v2_1_8/jpos/src/main/java/org/jpos/iso/GenericSSLSocketFactory.java#L142) for initialising the SSLContext.

> Also why do i need to have the server's certificate in the Keystore File, is the certificate there to be compared to the certificate sent by the server when the connection is invoked?

Yes, that is my understanding.

Is there something that is not working for you?

**Reply by nguyenthaichinghia** [reply]
Thanks for your help dear sir, currently everything is okay. I hope you have a great day!
